src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -71,6 +71,8 @@ public virtual IDisposable Start()
                 async features =>
                 {
                     var httpContext = contextFactory.CreateHttpContext(features);
+                    httpContext.Response.RegisterForDispose(httpContext);
```

Comment:
Going to return IDisposable from the closure; HttpContext has always been IDisposable, just never Disposed so will be just closing that loop.